Consumer key redirection support (fix #8)

thbkrkr opened the pull request:
Signed-off-by: thb thb.krkr@gmail.com

yadutaf commented on ovh/consumer_key.go:
```diff
@@ -40,6 +40,7 @@ type CkValidationState struct {
 type CkRequest struct {
 	client      *Client
 	AccessRules []AccessRule `json:"accessRules"`
+	Redirection string       `json:"redirection"`
```
Can you make this parameter optional ? https://api.ovh.com/1.0/auth.json

yadutaf commented:
Thanks for the fix! Can you add/fix related unit tests?

thbkrkr commented:
I have added 'omitempty' to make the parameter optional.
It fixes the unit tests. I don't know what pertinent unit test to add.
I have also fixed a typo in the test function names.

yadutaf commented:
Thanks ! I'd add a test to make sure ``NewCkRequestWithRedirection`` and then issuing a requests actually sends the redirection field

yadutaf commented:
Merged. Thanks !
